src/src/PluginStructs/P128_data_struct.h

```diff
@@ -2198,6 +2198,192 @@ const uint8_t PROGMEM ftv_gamma8[] = {
+           rgb, rrggbb,
+           rgb_tick_b = HtmlColor(0x505050),
+           rgb_tick_s = HtmlColor(0x101010),
+           rgb_m      = HtmlColor(0x00FF00),
+           rgb_h      = HtmlColor(0x0000FF),
+           rgb_s      = HtmlColor(0xFF0000);
+# endif // if defined(RGBW) || defined(GRBW)
+
+  int16_t fadedelay = 20;
+
+  uint16_t pixelCount = ARRAYSIZE;
+
+  int8_t defaultspeed  = 25,
+         rainbowspeed  = 1,
+         speed         = 25,
+         count         = 1,
+         rev_intensity = 3;
+
+  uint32_t _counter_mode_step = 0,
+           fadetime           = 1000,
+           ftv_holdTime,
+           pixelNum;
+
+  uint16_t ftv_pr = 0, ftv_pg = 0, ftv_pb = 0; // Prev R, G, B;
+  uint32_t ftv_totalTime, ftv_fadeTime, ftv_startTime, ftv_elapsed;
+  uint16_t ftv_nr, ftv_ng, ftv_nb, ftv_r, ftv_g, ftv_b, ftv_i, ftv_frac;
+  uint8_t  ftv_hi, ftv_lo, ftv_r8, ftv_g8, ftv_b8;
+
+  String colorStr,
+         backgroundcolorStr;
+
+  bool gReverseDirection = false;
+  bool rgb_s_off         = false;
+  bool fadeIn            = false;
+
+  byte cooling    = 50,
+       sparking   = 120,
+       brightness = 31;
+
+  unsigned long counter20ms = 0,
+                starttime[ARRAYSIZE],
+                starttimerb,
+                maxtime = 0;
+
+  P128_modetype mode     = P128_modetype::Off,
+                savemode = P128_modetype::Off,
+                lastmode = P128_modetype::Off;
+
+  void     fade(void);
+  void     colorfade(void);
+  void     wipe(void);
+  void     dualwipe(void);
+  void     faketv(void);
+  void     rainbow(void);
+  uint32_t Wheel(uint8_t pos);
+  void     kitt(void);
+  void     comet(void);
+  void     theatre(void);
+  void     scan(void);
+  void     dualscan(void);
+  void     twinkle(void);
+  void     twinklefade(void);
+  void     sparkle(void);
+
+  // Fire
+  unsigned long fireTimer;
+  RgbColor      leds[ARRAYSIZE];
+  void fire(void);
+
+  /// random number seed
+  uint16_t rand16seed; // = RAND16_SEED;
+  uint8_t  random8();
+  uint8_t  random8(uint8_t lim);
+  uint8_t  random8(uint8_t min,
+                   uint8_t lim);
+  uint8_t  qsub8(uint8_t i,
+                 uint8_t j);
+  uint8_t  qadd8(uint8_t i,
+                 uint8_t j);
+  uint8_t  scale8_video(uint8_t i,
+                        uint8_t scale);
+  void     Fire2012(void);
+  void     fire_flicker();
+  void     Plugin_128_simpleclock();
+  uint32_t rgbStr2Num(String rgbStr);
+  void     hex2rgb(String hexcolor);
+  void     hex2rrggbb(String hexcolor);
+  void     hex2rgb_pixel(String hexcolor);
+  void     NeoPixelSendStatus(struct EventStruct *eventSource);
+
+  const char *modeName[19] = {
+    "off",         "on",         "fade",           "colorfade",     "rainbow",           "kitt",       "comet",
+    "theatre",     "scan",       "dualscan",       "twinkle",       "twinklefade",       "sparkle",    "fire",
+    "fireflicker", "wipe",       "dualwipe",       "faketv",        "simpleclock"
+  };
```

Comment:
Missing F(), so better make it an array of __FlashStringHelper

Reply:
I did a nearly flat move of the existing code from the external repo to here, (have to explicitly put the credits somewhere).
There are plenty of improvements possible, and this is one of them, to be handled in the task 'Code Improvements' that I have planned 😉 
There are also several arrays of `ARRAYSIZE` (=300) that can probably be merged into 1 array, and calling `parseString()` 15 or more times in a row to retrieve the _same_ argument (3/4/5) to convert as a color, is probably also not a good idea.